web_src/js/features/comp/LabelEdit.ts

```diff
@@ -1,96 +1,81 @@
-    $('#label-modal-id').val($(this).data('id'));
-
-    const $nameInput = $('.edit-label .label-name-input');
-    $nameInput.val($(this).data('title'));
-
-    const $isArchivedCheckbox = $('.edit-label .label-is-archived-input');
-    $isArchivedCheckbox[0].checked = this.hasAttribute('data-is-archived');
+  const syncModalUi = () => {
+    const hasScope = nameHasScope(elNameInput.value);
+    elExclusiveField.classList.toggle('disabled', !hasScope);
+    const showExclusiveWarning = hasScope && elExclusiveInput.checked && elModal.hasAttribute('data-need-warn-exclusive');
+    toggleElem(elExclusiveWarning, showExclusiveWarning);
+    if (!hasScope) elExclusiveInput.checked = false;
+  };
 
-    const $exclusiveCheckbox = $('.edit-label .label-exclusive-input');
-    $exclusiveCheckbox[0].checked = this.hasAttribute('data-exclusive');
-    // Warn when label was previously not exclusive and used in issues
-    $exclusiveCheckbox.data('exclusive-warn',
-      $(this).data('num-issues') > 0 &&
-      (!this.hasAttribute('data-exclusive') || !isExclusiveScopeName($nameInput.val())));
-    updateExclusiveLabelEdit('.edit-label');
+  const showLabelEditModal = (btn:HTMLElement) => {
+    // the "btn" should contain the label's attributes by its `data-label-xxx` attributes
+    const form = elModal.querySelector<HTMLFormElement>('form');
+    elLabelId.value = btn.getAttribute('data-label-id') || '';
+    elNameInput.value = btn.getAttribute('data-label-name') || '';
+    elIsArchivedInput.checked = btn.getAttribute('data-label-is-archived') === 'true';
+    elExclusiveInput.checked = btn.getAttribute('data-label-exclusive') === 'true';
+    elDescInput.value = btn.getAttribute('data-label-description') || '';
+    elColorInput.value = btn.getAttribute('data-label-color') || '';
+    elColorInput.dispatchEvent(new Event('input', {bubbles: true})); // trigger the color picker
 
-    $('.edit-label .label-desc-input').val(this.getAttribute('data-description'));
+    // if label id exists: "edit label" mode; otherwise: "new label" mode
+    const isEdit = Boolean(elLabelId.value);
 
-    const colorInput = document.querySelector('.edit-label .js-color-picker-input input');
-    colorInput.value = this.getAttribute('data-color');
-    colorInput.dispatchEvent(new Event('input', {bubbles: true}));
+    // if a label was not exclusive but has issues, then it should warn user if it will become exclusive
+    const numIssues = parseInt(btn.getAttribute('data-label-num-issues') || '0');
+    elModal.toggleAttribute('data-need-warn-exclusive', !elExclusiveInput.checked && numIssues > 0);
+    elModal.querySelector('.header').textContent = isEdit ? elModal.getAttribute('data-text-edit-label') : elModal.getAttribute('data-text-new-label');
 
-    $('.edit-label.modal').modal({
+    const curPageLink = elModal.getAttribute('data-current-page-link');
+    form.action = isEdit ? `${curPageLink}/edit` : `${curPageLink}/new`;
```

Comment:
```suggestion
    form.action = `${curPageLink}/${isEdit ? 'edit' : 'new'}`;
```

Reply:
No, my code is easier to search "/edit" or "/new" globally.